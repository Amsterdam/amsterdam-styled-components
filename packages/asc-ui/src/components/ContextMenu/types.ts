enum Position {
  bottom = 'bottom',
  top = 'top',
}

export default {
  Position,
}
