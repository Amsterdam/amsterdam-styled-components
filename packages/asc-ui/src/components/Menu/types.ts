import { Theme } from '@datapunt/asc-core'

export namespace MenuStyleProps {
  export enum Position {
    bottom = 'bottom',
    top = 'top',
  }

  export enum Align {
    left = 'left',
    right = 'right',
  }

  export type SharedStyleProps = {
    focused?: boolean
    height?: number
  }

  export type MenuButtonStyleProps = {} & SharedStyleProps

  export type MenuItemStyleProps = {
    role?: string
    id?: string
    label?: string
    open?: boolean
    divider?: boolean
    hoverColor?: Theme.TypeLevel
  } & SharedStyleProps

  export type MenuListStyleProps = {
    labelId?: string
    position?: Position
    right?: number
    top?: number
    focused?: boolean
    alignRight?: boolean
    css?: string
  }

  export type MenuWrapperStyleProps = {
    css?: string
  } & SharedStyleProps
}
