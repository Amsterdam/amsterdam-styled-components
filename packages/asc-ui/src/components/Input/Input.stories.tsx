import React from 'react'
import { action } from '@storybook/addon-actions'
import Input from './Input'

export default {
  title: 'Experimental/Atoms/Input',

  decorators: [
    (storyFn: () => React.ReactNode) => (
      <div style={{ padding: '40px 10px' }}>{storyFn()}</div>
    ),
  ],
}

export const ControlledState = () => {
  const [text, setText] = React.useState('')

  return (
    <Input
      id="test-id"
      onBlur={action('onBlur called')}
      onChange={(e) => {
        setText(e.target.value)
        action(`onChange called: ${text}`)
      }}
      onFocus={action('onFocus called')}
      onKeyDown={action('onKeyDown called')}
      value={text}
    />
  )
}
