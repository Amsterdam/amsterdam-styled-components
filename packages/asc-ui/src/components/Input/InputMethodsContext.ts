import { createContext } from 'react'

const InputMethodsContext = createContext({})

export default InputMethodsContext
