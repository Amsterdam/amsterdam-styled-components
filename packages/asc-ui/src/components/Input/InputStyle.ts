import styled from '@datapunt/asc-core'
import { color } from '../../utils'

const InputStyle = styled.input.attrs({
  type: 'text',
  autoCapitalize: 'off',
  autoComplete: 'off',
  autoCorrect: 'off',
  spellCheck: false,
})`
  appearance: none;
  font-size: 1rem;
  border: solid 1px ${color('tint', 'level5')};
  border-radius: 0;
  box-sizing: border-box;
  padding: 10px 10px;
  width: 100%;
`

export default InputStyle
