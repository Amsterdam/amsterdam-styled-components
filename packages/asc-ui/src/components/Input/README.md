# Input
[Design System](https://designsystem.amsterdam.nl/7awj1hc9f/p/657629-text-input)

**Component status: experimental**