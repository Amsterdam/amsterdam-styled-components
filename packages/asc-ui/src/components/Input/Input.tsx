import React from 'react'
import InputStyle from '../../styles/components/InputStyle'

type InputProps = {
  placeholder?: string
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void
  onKeyDown: (event: React.KeyboardEvent<HTMLInputElement>) => void
  value: string
}

const Input: React.FC<InputProps> = ({
  placeholder,
  onChange,
  onKeyDown,
  ...otherProps
}) => (
  <InputStyle
    placeholder={placeholder}
    onChange={onChange}
    onKeyDown={onKeyDown}
    {...otherProps}
  />
)

export default Input
