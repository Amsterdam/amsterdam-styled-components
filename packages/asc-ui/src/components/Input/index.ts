import InputStyle from './InputStyle'
import InputContext from './InputMethodsContext'

export { default, InputProps, InputMethods } from './Input'

export const InputStyles = {
  InputStyle,
}

export { InputContext }
