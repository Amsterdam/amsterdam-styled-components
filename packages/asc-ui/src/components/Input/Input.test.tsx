import { render, fireEvent } from '@testing-library/react'
import { createRef } from 'react'
import { KeyboardKeys } from '../../types'
import Input from './Input'
import ThemeProvider from '../../theme/ThemeProvider'

jest.useFakeTimers()

describe('Input', () => {
  const onBlurMockFn = jest.fn()
  const onChangeMockFn = jest.fn()
  const onFocusMockFn = jest.fn()
  const onKeyDownMockFn = jest.fn()
  const props = {
    id: 'id',
    onBlur: onBlurMockFn,
    onChange: onChangeMockFn,
    onFocus: onFocusMockFn,
    onKeyDown: onKeyDownMockFn,
    value: 'test-value',
  }

  afterEach(() => {
    jest.resetAllMocks()
  })

  describe('component behaviour', () => {
    let input: any

    beforeEach(() => {
      const { container } = render(<Input {...props} />)
      input = container.firstChild
    })

    it('should trigger the onOnChange event', () => {
      fireEvent.change(input, { target: { value: 'a' } })
      expect(onChangeMockFn).toHaveBeenCalledTimes(1)
    })

    it('should trigger the submit when enter is pressed', () => {
      fireEvent.keyDown(input, { key: KeyboardKeys.Enter })
      expect(onKeyDownMockFn).toHaveBeenCalledTimes(1)
    })
  })

  it('should handle refs', () => {
    const ref = createRef<HTMLInputElement>()

    render(
      <ThemeProvider>
        <Input ref={ref} />
      </ThemeProvider>,
    )

    expect(ref.current).toBeInstanceOf(HTMLInputElement)
    expect(ref.current?.type).toEqual('text')
  })
})
