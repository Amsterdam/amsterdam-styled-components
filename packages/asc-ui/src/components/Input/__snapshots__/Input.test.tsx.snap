// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component structure should render 1`] = `
<input
  autocapitalize="off"
  autocomplete="off"
  autocorrect="off"
  class="bianDr"
  id="id"
  spellcheck="false"
  type="text"
  value="test-value"
/>
`;
